akka-remote/src/main/scala/akka/remote/security/setup/CryptoServiceProviderSetup.scala

```diff
@@ -0,0 +1,33 @@
+package akka.remote.security.setup
+
+import java.security.Provider
+import javax.net.ssl.ManagerFactoryParameters
+
+import akka.actor.setup.Setup
+
+// TODO should be one Setup per Provider, with a common trait
```

Comment:
Multiple `Setup`, with one `Provider` and optional `Parameter` would make it more intuitively obvious what the `Provider` is used for, and make the approach more easily extensible.

I don't think it makes sense to allow a `Parameter` without a provider and the implementation of the former is entirely dependent on the latter and is in no way type safe. In particular `SunX509` does not support any `Parameter`, and would throw if called with one.